### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by immerrr**
This PR adds pytest-catchlog for reviewing errors sent to logs, e.g. from background threads.

It also adds a rather unique frontier name for each test case so that test case ordering doesn't affect the vcrpy cassettes.

### CODE REVIEW

**Comment by immerrr** [comment]
Sweet, I like them, merged your changes into my branch right away :)

File: tests/hubstorage/test_frontier.py [hunk 1]
```diff
@@ -3,126 +3,125 @@
...
     fps3 = [{'fp': '/index_%s.html' % fp}
             for fp in range(batch_size * 2, batch_size * 3)]
-    frontier.add(TEST_FRONTIER_NAME, TEST_FRONTIER_SLOT, fps3)
+    frontier.add(frontier_name, TEST_FRONTIER_SLOT, fps3)
     frontier.flush()
 
     assert frontier.newcount == 150 + old_count
 
     # insert repeated fingerprints
-    fps4 = [{'fp': '/index_%s.html' % fp} for fp in range(0, batch_size)]
```
Comment (vshlapakov) [inline tests/hubstorage/test_frontier.py]: @immerrr last question about the PR before merge, why the test with repeated fps was changed?

Comment (immerrr) [reply tests/hubstorage/test_frontier.py]: Because `fps4` was unused and `fps3` were used as "repeated fingerprints" instead.

Comment (vshlapakov) [reply tests/hubstorage/test_frontier.py]: Oh, thanks for cleaning it, I didn't spot it even now :man_facepalming:

**Comment by vshlapakov** [comment]
Thanks again @immerrr! :+1:

**Pull request merged**